kvbc/test/categorization/kv_blockchain_test.cpp

```diff
@@ -478,6 +476,159 @@ TEST_F(categorized_kvbc, link_state_transfer_chain) {
   }
 }
 
+TEST_F(categorized_kvbc, creation_of_category_type_cf) {
+  KeyValueBlockchain block_chain{db, true};
+  ASSERT_TRUE(db->hasColumnFamily(detail::ST_CHAIN_CF));
+}
+
+TEST_F(categorized_kvbc, instantiation_of_categories) {
+  auto wb = db->getBatch();
+  db->createColumnFamily(detail::CAT_ID_TYPE);
+  wb.put(detail::CAT_ID_TYPE, std::string("merkle"), std::string(1, static_cast<char>(detail::CATEGORY_TYPE::merkle)));
+  wb.put(detail::CAT_ID_TYPE, std::string("merkle2"), std::string(1, static_cast<char>(detail::CATEGORY_TYPE::merkle)));
+  wb.put(detail::CAT_ID_TYPE, std::string("imm"), std::string(1, static_cast<char>(detail::CATEGORY_TYPE::immutable)));
+  wb.put(detail::CAT_ID_TYPE, std::string("kvhash"), std::string(1, static_cast<char>(detail::CATEGORY_TYPE::kv_hash)));
+
+  db->write(std::move(wb));
+
+  KeyValueBlockchain block_chain{db, true};
+  KeyValueBlockchain::KeyValueBlockchain_tester tester;
+  auto cats = tester.getCategories(block_chain);
+  ASSERT_EQ(cats.count("merkle"), 1);
+  ASSERT_THROW(std::get<KVHashCategory>(cats["merkle"]), std::bad_variant_access);
+  ASSERT_NO_THROW(std::get<MerkleCategory>(cats["merkle"]));
+
+  ASSERT_EQ(cats.count("merkle2"), 1);
+  ASSERT_THROW(std::get<KVHashCategory>(cats["merkle2"]), std::bad_variant_access);
+  ASSERT_NO_THROW(std::get<MerkleCategory>(cats["merkle2"]));
+
+  ASSERT_EQ(cats.count("imm"), 1);
+  ASSERT_THROW(std::get<KVHashCategory>(cats["imm"]), std::bad_variant_access);
+  ASSERT_NO_THROW(std::get<detail::ImmutableKeyValueCategory>(cats["imm"]));
+
+  ASSERT_EQ(cats.count("kvhash"), 1);
+  ASSERT_THROW(std::get<MerkleCategory>(cats["kvhash"]), std::bad_variant_access);
+  ASSERT_NO_THROW(std::get<KVHashCategory>(cats["kvhash"]));
+}
+
+TEST_F(categorized_kvbc, throw_on_non_exist_category_type) {
+  KeyValueBlockchain block_chain{db, true};
+  auto wb = db->getBatch();
+  wb.put(detail::CAT_ID_TYPE, std::string("merkle"), std::string(1, static_cast<char>(4)));
```

Comment:
Maybe we can change the hardcoded value of 4 to some special enum value (i.e. last + 1, invalid cat ID or something like that) in order make it more explicit and reduce the risk of this test failing at some point.

Reply:
true